File: polyglot/6-exceptions-threads-asyncs.py

# try-catch-finally
foo: int = 1
try:
    raise Exception('test')
except TypeError:
    assert False
except SyntaxError:
    assert False
except Exception as e:
    assert isinstance(e, Exception)
    assert str(e) == 'test'
    foo += 1
finally:
    foo **= 2

assert foo == 4

# threads
"""TODO"""

# asyncio
import asyncio
from typing import Awaitable

async def main () -> None:
    print('main:', 'start...')
    await asyncio.sleep(1)
    print('main:', '1 sec.')

asyncio.run(main())

File: polyglot/6-exceptions-threads-asyncs.ts
import assert from 'assert'

/* try-catch-finally */
let foo = 1
try {
  throw new Error('test')
} catch (e) {
  assert(e instanceof Error)
  assert(e.message === 'test')
  foo += 1
} finally {
  foo **= 2
}

assert(foo === 4)

/* async/await */
async function main (): Promise<void> {
  console.log('main:', 'start...')
  await new Promise(function (resolve) {
    setTimeout(resolve, 1000)
  })
  console.log('main:', '1 sec')
}

void main()
